File: backend/controllers/movementTitle.js
const MovementTitle = require('../models/MovementTitle');

async function getMovementTitle (req, res) {
    try{
        const mt = await MovementTitle.find();

        if(mt) {
            res.status(200).json({ success: true, data: mt })
        } else {
            res.status(404).json({ success: false, message: 'Movement title not found!' });
        }
    } catch(error) {
        console.error(error);
        res.status(500).json({ error: 'Internal server error!' });
    }
}

async function addMovementTitle (req, res) {
    try{
        const { t_name } = req.body;

        const mt = new MovementTitle({
            t_name: t_name,
            date_time: Date.now(),
            is_active: true
        });

        const savedMT = await mt.save();
        if(savedMT) {
            res.status(201).json({ success: true, data: savedMT });
        } else {
            res.status(400).json({ success: false, message: 'Movement title error!' })
        }
    } catch(error) {
        console.error(error);
        res.status(500).json({ message: error });
    }
}

async function updateMovementTitle (req, res) {
    try{
        const mt_id = req.params.id;
        const { t_name } = req.body;

        const mt = await MovementTitle.findById(mt_id);
        if(!mt) {
            return res.status(404).json({ success: false, message: 'Movement title not found!' });
        }

        mt.t_name = t_name;
        mt.save();

        res.status(200).json({ success: true, message: 'Movement title updated successfully' });
    } catch(error) {
        console.error(error);
        res.status(500).json({ message: error });
    }
}

async function deleteMovementTitle (req, res) {
    try{
        const mt_id = req.params.id;
        const mt = await MovementTitle.findById(mt_id);

        if(!mt) {
            res.status(404).json({ success: false, message: 'Movement title not found!' });
        } else {
            await mt.updateOne({ is_active: false });
            res.status(200).json({ success: true, message: 'Movement title deleted successfully.' });
        }
    } catch(error) {
        console.error(error);
        res.status(500).json({ error: 'Internal server error!' });
    }
}

async function getMovementTitleById (req, res) {
    try{
        const mt_id = req.params.id;
        const mt = await MovementTitle.findById(mt_id);

        if(mt && mt.is_active === true){
            res.status(200).json({ success: true, data: mt });
        } else {
            res.status(404).json({ success: false, error: 'Movement title not found!' });
        }
    } catch(error) {
        console.error(error);
        res.status(500).json({ error: 'Internal server error!' });
    }
}

module.exports = {
    getMovementTitle,
    addMovementTitle,
    updateMovementTitle,
    deleteMovementTitle,
    getMovementTitleById
}

File: backend_python/models/movementTitle.py
from bson import ObjectId

class MovementTitle:
    def __init__(self, db):
        self.collection = db['movementTitle']
    
    def get_all_title(self):
        try:
            titles = list(self.collection.find({}))
            print("Movement title fetched from db:", titles)
            for title in titles:
                title['_id'] = str(title['_id'])
            return titles
        except Exception as e:
            print("Error fetching movement:", str(e))
            return []
    
    def add_title(self, t_name):
        try:
            title = {
                "t_name": t_name,
                "date_time": "today",
                "is_active": True
            }
            self.collection.insert_one(title)
        except Exception as e:
            raise e
    
    def update_title(self, mt_id, t_name):
        try:
            self.collection.update_one({"_id": ObjectId(mt_id)}, {"$set": {"t_name": t_name}})
        except Exception as e:
            raise e
    
    def delete_title(self, mt_id):
        try:
            self.collection.delete_one({"_id": ObjectId(mt_id)})
        except Exception as e:
            raise e